llmfoundry/data/finetuning/tasks.py

```diff
@@ -306,7 +314,18 @@ def dataset_mapper(example: Dict):
                 f'Dropped {examples_removed} examples where the prompt was longer than {max_seq_len}.'
             )
 
-        return prompt_length_filtered_dataset
+        empty_examples_dropped_dataset = prompt_length_filtered_dataset.filter(
+            lambda example: len(example['input_ids']) > 0 and len(example[
+                'labels']) > 0 and not all(token_id == tokenizer.pad_token_id
```

Comment:
If `tokenizer.pad_token_id` is not set yet will this be a problem? (I'm guessing no because I think that it will just return `None` but worth considering.)

And should there be a similar check for EOS? Maybe if all the label tokens are EOS or PAD, that should be filtered.

Reply:
@alextrott16 the critical issue was that if it was all PAD, the collate fn strips pads and then you end with empty target and it crashes. This is not true if its all EOS/BOS, it will train, its just kind of a weird example. wdyt?

Reply:
Filtering error-causing examples is definitely a must, and that base is covered. The rest is a judgement call. And maybe you're right that it is better to avoid trying to catch examples that just seem weird. I guess I don't have any super strong feelings there, so I am happy to go with your judgement call.